src/analysis/sampling/CHRR/chrrParseModel.m

```diff
@@ -1,34 +1,29 @@
-function P = chrrParseModel(model)
-% Parses a COBRA model into the right format for the CHRR sampler
-%
-% USAGE:
-%
-%    P = chrrParseModel(model);
-%
+function [P,model] = chrrParseModel(model)
+% CHRRPARSEMODEL Parse a COBRA model into the right format for the CHRR sampler
+% 
+% [P,model] = chrrParseModel(model);
+% 
+% We are trying to sample uniformly at random from the points v that satisfy:
+%            Sv = b
+%       lb <= v <= ub
+% 
 % INPUTS:
-%    model:      COBRA model structure with fields:
-%
-%                 * .S - The `m x n` stoichiometric matrix
-%                 * .lb - `n x 1` lower bounds on fluxes
-%                 * .ub - `n x 1` upper bounds on fluxes
-%                 * .c - `n x 1` linear objective
-%
-% OUTPUT:
-%    P:          A structure with fields:
-%
-%                 * .A_eq - Equality constraint matrix (`model.S`)
-%                 * .b_eq - Right hand side of equality constraints (`model.b`)
-%                 * .A - Inequality constraint matrix ([:math:`I_n 0; 0 -I_n`])
-%                 * .b - Right hand side of inequality constraints ([`lb`; `-ub`])
-%
-% .. Authors: - Ben Cousins and Hulda S. Haraldsdóttir, October 2016
-%
-% We are trying to sample uniformly at random from the points `v` that satisfy:
-%
-% .. math::
-%    min  ~&~ c v \\
-%    s.t. ~&~ S v = b \\
-%         ~&~ lb \leq v \leq ub
+% model ... COBRA model structure with fields:
+% .S ... The m x n stoichiometric matrix
+% .lb ... n x 1 lower bounds on fluxes
+% .ub ... n x 1 upper bounds on fluxes
+% 
+% OUTPUTS:
+% P ... A structure with fields:
+% .A_eq ... Equality constraint matrix (model.S)
+% .b_eq ... Right hand side of equality constraints (model.b)
+% .A ... Inequality constraint matrix ([I_n 0; 0 -I_n])
+% .b ... Right hand side of inequality constraints ([lb; -ub])
+% 
+% October 2016, Ben Cousins and Hulda S. Haraldsdóttir
+%
+% --Moved objective function handling to preprocess function: Ben Cousins,
+% November 2017

```

Comment:
Please format the documentation according to the [documentation guide](https://github.com/opencobra/cobratoolbox/blob/master/docs/source/guides/documentationGuide.md)